### PR TITLE
chore: fixing file lock issues on autocommit

### LINKED ISSUE #33384: Task: remove redundant call to migrate dsl from the consolidated Api, it's causing file lock issues.

### PR DESCRIPTION
**Pull request by sondermanish**
## Description 
- Refactoring consolidated Api to replace autocommit trigger.
- Reduced # db-calls for new pages.

Fixes #33384 

## Automation

/ok-to-test tags="@tag.All"

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Added methods for retrieving and constructing application pages based on branch and mode.
  - Improved logic and error handling for fetching application pages.
  - Enhanced schema migration for Git-connected apps.

- **Refactor**
  - Replaced method calls in `ConsolidatedAPIServiceImpl` for better performance.
  - Modified method in `ApplicationServiceCE` for finding applications by default ID, branch name, and mode.

- **Tests**
  - Updated test cases in `ConsolidatedAPIServiceImplTest` for method call modifications.

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9110390548>
> Commit: a175879d86edbb9b221ac41570d76c0fc1ca754d
> Cypress dashboard: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9110390548&attempt=3&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank"> Click here!</a>
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts </ol>
> To know the list of identified flaky tests - <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">Refer here</a>

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ConsolidatedAPIServiceImpl.java [hunk 1]
```diff
@@ -199,14 +203,31 @@ public Mono<ConsolidatedAPIResponseDTO> getConsolidatedInfoForPageLoad(
...
 
-        /* Get all pages in application */
-        Mono<ResponseDTO<ApplicationPagesDTO>> applicationPagesDTOResponseDTOMonoCache = applicationIdMonoCache
-                .flatMap(appId -> newPageService.findApplicationPages(appId, null, branchName, mode))
+        // dslMigration-over-here using the branchName and defaultId
+        Mono<Application> branchedApplicationMonoCached = applicationIdMonoCache
+                .flatMap(defaultApplicationId -> applicationService.findByDefaultIdBranchNameAndApplicationMode(
+                        defaultApplicationId, branchName, mode))
+                .cache();
+
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ConsolidatedAPIServiceImpl.java:211]: it already has error handling

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java [hunk 2]
```diff
@@ -307,6 +421,15 @@
...
     }
 
+    @Override
+    public Mono<PageDTO> getPageAndMigrateDslByBranchAndDefaultPageId(
+            String defaultPageId, String branchName, boolean viewMode, boolean migrateDsl) {
+        // Fetch the page with read permission in both editor and in viewer.
+        return newPageService
+                .findByBranchNameAndDefaultPageId(branchName, defaultPageId, pagePermission.getReadPermission())
+                .flatMap(newPage -> getPageDTOAfterMigratingDSL(newPage, viewMode, migrateDsl));
+    }
```
[the remark this answers is not in the document]
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationPageServiceCEImpl.java:383]: If the migration fails then error shoulnd't be handled for db page migration

**Review by nidhi-nair — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/applications/base/ApplicationServiceCEImpl.java [hunk 3]
```diff
@@ -884,7 +884,7 @@ public Mono<String> findBranchedApplicationId(
                         AppsmithError.NO_RESOURCE_FOUND,
                         FieldName.APPLICATION,
                         defaultApplicationId + ", " + branchName)))
-                .map(Application::getId);
+                .map(application -> application.getId());
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/applications/base/ApplicationServiceCEImpl.java:887]: Why was this change required?

Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/applications/base/ApplicationServiceCEImpl.java:887]: The EE and CE had this diff, hence chose to cover this in the same pr.

**Pull request merged**